src/arcana-linter/WarningLog.hh

```diff
@@ -155,8 +51,9 @@ private:
     /// The amount of headers without #pragma once
     int mNumMissingPragmaOnce = 0;
 
-    /// The amount of parse errors
     int mNumParseErrors = 0;
+
+    int mNumMissingNoexcept = 0;
 };
 
 }
```

Comment:
this seems like a cumbersome system to add future warnings.

maybe have a virtual "Diagnostic" class that all the diagnostics derive from. They internally store a list of affects files/lines.

Ideally adding a new diagnostic would require a single new class (.cc + .hh) and maybe a registration call